### PR TITLE
gcylc: dot view - maintain user selection during update

### LINKED ISSUE #1707: cylc gui: dot view redraw resetting selection
Somewhat frustratingly for users with large suites using the dot view of cylc gui, it seems that whenever a redraw happens the view gets reset so you are no longer focused on the area of interest.

@benfitzpatrick - I seem to remember we had a similar problem with treeview, can you advise?

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1707

Doesn't go as far as Ben's incremental build suggestion (https://github.com/cylc/cylc/issues/1707#issuecomment-167763411).

### CODE REVIEW

File: lib/cylc/gui/updater_dot.py [hunk 1]
```diff
@@ -303,9 +304,39 @@ def _get_expanded_rows(self):
...
+            self.selected_rows.append(model.get_value(model.get_iter(path), 0))
+
+    def _set_selected_rows(self):
+        """Re-Selects previously selected rows where possible.
+
+        Uses self.selected_rows to determine which rows to select.
+        Warning: This method has not been tested with multiple selection.
+
+        """
+        def should_select_row(model, path, iter_, selection):
```
Comment (matthewrmshin) [inline lib/cylc/gui/updater_dot.py]: Any reason why this function/method needs to be embedded?

Comment (oliver-sanders) [inline lib/cylc/gui/updater_dot.py]: The logic doesn't really make sense within DotUpdater so I encapsulated it (pep227). I could convert it to a lambda if you prefer.

**Comment by hjoliver** [comment]
(sorry - didn't notice I was not assigned as a reviewer :grim)

File: lib/cylc/gui/updater_dot.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/cylc/gui/updater_dot.py]: (The method, however, does refer to `self`.)

Comment (arjclark) [inline lib/cylc/gui/updater_dot.py]: Once this is resolved happy for this to be merged.

Comment (oliver-sanders) [inline lib/cylc/gui/updater_dot.py]: @matthewrmshin Code now moved into a static method.

**Comment by matthewrmshin** [comment]
Doubly approved already, so I think we can remove @arjclark as reviewer.

**Pull request merged**